src/Square3x3.java

```diff
@@ -0,0 +1,206 @@
+        if (validIndex(row) && validIndex(col))
+            return square[row][col];
+        return -1;
+    }
+
+    /**
+     * Sets the value in [row][col] cell of the Square3x3 object
+     *
+     * @param row   - the row of the Square3x3
+     * @param col   - the col of the Square3x3
+     * @param value - the new value to set in [row][col] cell
+     */
+    public void setXY(int row, int col, int value) {
+        if (validIndex(row) && validIndex(col))
+            square[row][col] = value;
+    }
+
+    // methods
+
+    // Check if index is valid for Square3x3 object
+    private boolean validIndex(int index) {
+        return 0 <= index && index < SIZE;
+    }
+
+    // Create a new Square3x3 object. Initialize all cells value with the given value parameter
+    private void initSquare(int value) {
+        square = new int[SIZE][SIZE];
+
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                this.square[i][j] = value;
+            }
+        }
+    }
+
+    // Initialize WHICH_NUMBERS_APPEAR boolean array
+    // [0] item value with true
+    // [1] - [9] items value with false
+    private void initWhichNumbersAppear() {
+        WHICH_NUMBERS_APPEAR[0] = true;
+        for (int i = 1; i < WHICH_NUMBERS_APPEAR.length; i++) {
+            WHICH_NUMBERS_APPEAR[i] = false;
+        }
+    }
+
+    /**
+     * Returns a string representation of the Square3x3 object
+     *
+     * @return a string represent the Square3x3 object, For example:
+     * 1   2   3
+     * 4   5   6
+     * 7   8   9
+     */
+    public String toString() {
+        String msg = "";
+        for (int i = 0; i < SIZE; i++) {
+            for (int j = 0; j < SIZE; j++) {
+                msg += square[i][j];
+                if (j < SIZE - 1)
+                    msg += "\t";
+            }
+            msg += "\n";
+        }
+        return msg;
+    }
+
+    // print a row in square
+    //TODO: delete - only for tests
+    public void printRow(int row) {
+        for (int col = 0; col < SIZE; col++)
+            System.out.print(getCell(row, col) + "\t");
+    }
+
+    /**
+     * Returns true if all numbers 1-9 exists in square
+     *
+     * @return true if all numbers 1-9 exists in square
+     */
+    public boolean allThere() {
+        initWhichNumbersAppear(); // init WHICH_NUMBERS_APPEAR boolean array
+
+        // go over all square rows
+        for (int i = 0; i < SIZE; i++) {
+            this.whosThereRow(i, WHICH_NUMBERS_APPEAR); // check whosThereRow for each row
+        }
+
+        // go over WHICH_NUMBERS_APPEAR boolean array
+        for (int j = 0; j < WHICH_NUMBERS_APPEAR.length; j++) {
+            if (!WHICH_NUMBERS_APPEAR[j])
+                return false; // a number between 1-9 is not in square
+        }
+        return true; // numbers 1-9 are all in square
+    }
+
+    /**
+     * Check if square row contains cells with values between 1-9
```

Comment:
when you describe a method that manipulates a value, you can't start a sentence with a "check".
a better description starts as: `fill the 'values' array with "trues"... `

This is an important practice, because it is very important to emphasize that a function actually change values, and not just verifies something.

BTW, the description is ok! this is a minor comment, so other developers will use your method carefully.